### PR TITLE
2021 wohngeld reform

### PR DESCRIPTION
**Pull request by Eric-Sommer**
Implements a wohngeld/housing benefit reform becoming effective 01/2021. Before, only rent exluding heating costs (Kaltmiete) were considered for the housing benefit. From 2021, heating costs are considered via a lump sum amount depending on household size.

The reform is tested and confirmed with [smart-rechner.de](https://www.smart-rechner.de/wohngeld/rechner.php). gettsim returns €283.60, which is the last step before rounding to full euros (see #88).
grafik

### CODE REVIEW

**Comment by hmgaudecker** [comment]
Thanks -- so how much is missing here? Thinking about releasing the 2021 version.

**Comment by Eric-Sommer** [comment]
For starters, the code does not run atm. I also discussed privately with @mjbloemer on what the reform actually does. We'd certainly need an additional test here. I'll have a look at it later tonight.

**Comment by Eric-Sommer** [comment]
Should be fine now. Rent considered after is aligned with external sources, but there is a more general issue with relevant income, which will be addressed in a separate issue.

File: gettsim/transfers/wohngeld.py [hunk 1]
```diff
@@ -489,7 +489,7 @@ def wohngeld_miete_ab_2009(
     return wg_miete
 
 
-def wohngeld_max_miete_ab_2021(
+def wohngeld_miete_ab_2021(
```
Comment (hmgaudecker) [inline gettsim/transfers/wohngeld.py:492]: Docstring still says 2009. 

Why are there no tests for this set of functions? Seems reasonable easy that an error would slip in...

Given that the PR starts with that calculator, taking a couple of those examples including extreme ones / corner cases (hh_size in {5, 6, 13}) and making test cases out of them would seem like the natural place to start.

Comment (Eric-Sommer) [reply gettsim/transfers/wohngeld.py:492]: Docstring is corrected.

There are two tests for 2021. The one I describe above and `test_increasing_hh_size()` is extended for 2021. To me, this confirms that the logic of the 2021 reform is captured correctly as it does differs only in the maximum value for the considered rent. Extreme cases (hh_size > 12) are an issue that would affect all wohngeld calculations and would deserve an own PR in my view. @mjbloemer has played around with wohngeld calculations and very large households.

**Comment by hmgaudecker** [comment]
Okay, I was just surprised that the renaming of the function did not lead
to renaming anywhere in the tests.

In the long run, it would be good to also test functions like this
directly, but no prerequisite for merging this IMO.

Eric-Sommer <notifications@github.com> schrieb am So., 21. Feb. 2021, 10:51:

File: gettsim/transfers/wohngeld.py (hunk 1 above)
Comment (hmgaudecker) [reply gettsim/transfers/wohngeld.py:492]: Okay, I was just surprised that the renaming of the function did not lead
to renaming anywhere in the tests.

In the long run, it would be good to also test functions like this
directly, but no prerequisite for merging this IMO.

Comment (hmgaudecker) [reply gettsim/transfers/wohngeld.py:492]: Testing each function directly will also make it easier to catch side-effects, see #272

**Review by hmgaudecker — APPROVED** [review]
Looks good to me from the code side, thanks! 

However, @mjbloemer, it would be great if you could have a quick look at the parameters.

**Comment by mjbloemer** [comment]
> Looks good to me from the code side, thanks!
> 
> However, @mjbloemer, it would be great if you could have a quick look at the parameters.

parameters (heating cost) are fine. You could add the heating cost parameters for 2009 and 2010 which followed the same rule.

I have not looked at the calculations/structure.

**Comment by hmgaudecker** [comment]
Great, feel free to merge and we will [tackle the tests](https://gettsim.zulipchat.com/#narrow/stream/212222-general/topic/Wohngeld/near/227247729) in a separate PR.
